src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -148,10 +148,21 @@ public SBApiResponse profileUpdate(Map<String, Object> request, String userToken
 			List<String> approvalFieldList = approvalFields();
 			String newDeptName = checkDepartment(profileDetailsMap);
 			Map<String, Object> transitionData = new HashMap<>();
-			for (String approvalList : approvalFieldList) {
-				if (profileDetailsMap.containsKey(approvalList)) {
-					transitionData.put(approvalList, profileDetailsMap.get(approvalList));
-					profileDetailsMap.remove(approvalList);
+			String mainKey="";
+			for (String approvalField : approvalFieldList) {
+				String[] fieldParts = approvalField.split("\\.");
+				if (fieldParts.length == 2) {
+					mainKey = fieldParts[0];
+					String subKey = fieldParts[1];
+					if (profileDetailsMap.containsKey(mainKey)) {
+						List<Map<String, Object>> nestedList = (List<Map<String, Object>>) profileDetailsMap.get(mainKey);
```

Comment:
everything inside profileDetails is not arrayList... 
this should be treated properly... better to have new private method to find the given key exists... 
new method should check the following.. 
1. there may be possible to add keys at the single level (without dots)... should handle that too... 
2. if keys are added with two level - retrieve the 1st level and check the type - based on 1st level type - access the list or map and then check the 2nd level key exists...